src/services/experiment-service.js

```diff
@@ -0,0 +1,113 @@
+import { EXPERIMENT_ENDPOINT, API_TIMEOUT } from '../global-constants';
+const axios = require('axios');
+
+let RETRY_LIMIT = 10;
+
+async function _getExperimentPercent(storeURL, account, experimentID) {
+    const url = `${storeURL}${EXPERIMENT_ENDPOINT}?key=${experimentID}`;
+    try {
+        const response = await axios({
+            url: url,
+            method: 'GET',
+            timeout: API_TIMEOUT,
+            headers: { 'Accept': 'application/json', 'Content-Type': 'application/json' },
+            auth: { username: account.email_address, password: account.data.auth_token },
+        });
+
+        return response.data || 0;
+    } catch (error) {
+        /**
+         * When tried to set an experiment too soon after a store is created
+         * we might get a 509. Wait for n seconds and retry a few times
+         */
+        if (error.response.status === 509) {
+            console.log('...');
+            if (RETRY_LIMIT < 1) {
+                _logMessage(`Failed to retrieve config percent for experiment ${experimentID}`);
+                throw error;
+            }
+            await new Promise(r => setTimeout(r, 3000));
+            RETRY_LIMIT--;
+            return await _getExperimentPercent(url, account, experimentID);
```

Comment:
It will be cleaner to change this method to take an optional `retries` parameter which defaults to your max. The way you have it would be problematic for parallelization because multiple threads might access `RETRY_LIMIT`

```suggestion
            if (retries < 1) {
                _logMessage(`Failed to retrieve config percent for experiment ${experimentID}`);
                throw error;
            }
            await new Promise(r => setTimeout(r, 3000));
            retries--;
            return await _getExperimentPercent(url, account, experimentID, retries);
```